Source/GSThemeDrawing.m

```diff
@@ -3678,85 +3593,52 @@ - (void) drawCellViewRow: (NSInteger)rowIndex
       return;
     }
   
-  // Check the delegate method...
-  hasMethod = [delegate respondsToSelector: @selector(outlineView:viewForTableColumn:item:)];
-  if (hasMethod)
-    {
-      ov = (NSOutlineView *)v;
-      outlineTableColumn = [ov outlineTableColumn];
-    }
-  else
-    {
-      hasMethod = [delegate respondsToSelector: @selector(tableView:viewForTableColumn:row:)];
-    }
-
   [self _calculatedStartingColumn: &startingColumn
 		     endingColumn: &endingColumn
-		    withTableView: v
+		    withTableView: tableView
 		       inClipRect: clipRect];
   
   /* Draw the row between startingColumn and endingColumn */
-  for (i = startingColumn; i <= endingColumn; i++)
+  for (columnIndex = startingColumn; columnIndex <= endingColumn; columnIndex++)
     {
-      NSRect drawingRect = [v frameOfCellAtColumn: i
-					      row: rowIndex];
-      NSTableColumn *tb = [tableColumns objectAtIndex: i];
-      NSIndexPath *path = [NSIndexPath indexPathForItem: i
-					      inSection: rowIndex];
-      NSView *view = [v _renderedViewForPath: path];
-
-      if (ov != nil)
+      id rowView = [tableView rowViewAtRow: rowIndex
+			   makeIfNecessary: YES];  
+      NSView *view = [tableView viewAtColumn: columnIndex
+					 row: rowIndex
+			     makeIfNecessary: YES];
+      
+      // If the view is already part of the table, don't re-add it...
+      if (rowView != nil
+	  && [[tableView subviews] containsObject: rowView] == NO)
 	{
-	  id item = [ov itemAtRow: rowIndex];
+	  NSRect cellFrame = [tableView frameOfCellAtColumn: 0
+							row: rowIndex];
+	  CGFloat x = 0.0;
+	  CGFloat y = cellFrame.origin.y;
+	  CGFloat w = [tableView frame].size.width;
+	  CGFloat h = [tableView rowHeight];
 	  
-	  if (tb == outlineTableColumn)
-	    {
-	      drawingRect = [self _drawOutlineTableColumn: tb
-					      outlineView: ov
-						     item: item
-					      drawingRect: drawingRect
-						 rowIndex: rowIndex];	      
-	    }
+	  NSRect rvFrame = NSMakeRect(x, y, w, h);
+	  NSAutoresizingMaskOptions options = NSViewWidthSizable
+	    | NSViewMinYMargin;
 	  
-	  if (view == nil)
-	    {
-	      if (hasMethod)
-		{
-		  view = [delegate outlineView: ov
-			    viewForTableColumn: tb
-					  item: item];
-		}
-	      else
-		{
-		  view = [self _prototypeCellViewFromTableColumn: tb];
-		}
-	    }	  
+	  [tableView addSubview: rowView];
+	  [rowView setAutoresizingMask: options];
+	  [rowView setFrame: rvFrame];
 	}
-      else
+      
+      // Create the view if needed...
+      if ([[rowView subviews] containsObject: view] == NO
```

Comment:
Again these conditions should be tested the other way around.

Reply:
Done